patterns/behavioural/command.md

```diff
@@ -0,0 +1,226 @@
+# Command
+
+## Description
+
+Suppose we have a sequence of actions or transactions
+encapsulated as objects.
+We want these actions or commands to be
+executed or invoked in some order later at different time.
+These commands may also be triggered as a result
+of some event. For example when a user pushes a button
+or on arrival of a data packet.
+In addition, these commands might be be undoable.
+For example, operations of the editor.
+We also may want to store logs of executed commands
+so that we could reapply the changes later if system crashes.
+
+## Motivation
+
+Define two database operations
+`create table` and `add field`.
+Each of these operations is a command
+which knows how to undo the command, e.g.,
+`drop table` and `remove field`.
+When a user invokes a database migration
+operation then each command is executed in the defined order,
+and when the user invokes the rollback operation
+then each undo operation is invoked in reverse order.
```

Comment:
This doesn't say for me "Why and where you should use the pattern" but more describes the actions that are done in our example?

Reply:
@simonsan It describes a concrete example. But I could add more general description of why and when we should use this pattern in a couple of sentences. In fact, I already mention in the Description part.

Reply:
yeah, i would propose to do that so people used to this template can easily check in `Motivation` why and where to use this pattern. maybe also to shorten then the description and move the corresponding sentences to motivation bringing the text that is now in motivation to an `# Example` headline as a description of the following approaches

Reply:
@simonsan Made some changes.